### PR TITLE
B1 Drivers Home Page - Suf

### PR DESCRIPTION
**Pull request by sufka861**
My assignment was to complete the screen B1 - Drivers Home Page.

### **Screens & Components**
- In the (src/screens/MyCar) App.tsx - I added a "Drawer.Screen" named "DriversHomeScreen" which leads to the screen.
- In the src/screens/MyCar directory I added the DriversHomeScreen.js which includes the content of the screen.
- In the src/screens/MyCar I added a directory called "b1Components"
**Components under b1Components:**
1. ServiceCard.js - card component for each service in the top vertical scrollbar.
2. MessageCard.js - card component for each message in the middle vertical scrollbar.
    - Each message has a description. The component has a truncateDescription function to handle the display of long descriptions. If the description length is longer then MAX_DESCRIPTION_WORDS => the displayed description would consist of the first MAX_DESCRIPTION_WORDS followed by '...'.
4. BigServiceCard.js - card component for each major (big) service listed in the bottom of the screen.

### **Assets**
- In the src/assets directory I added the new B1-Assets directory which holds all of the image assets of the screen.
- DriversHomeScreenText.json - holds hebrew text that was used in the DriversHomeScreen
- serviceGalleryData.js - holds an array of the galleryData used to produce the service scrollbar
- messagesGalleryData.js - holds an array of the messagesGalleryData used to produce the messages scrollbar
   - If the array is empty, the component would not show at all
   - If an item in the array does not have an image path, a default image will display

### **Navigation**

- Each component that should have the functionality of being pressable and to navigate to a different screen, is implemented with a `<TouchableOpacity>` tag.
- The onPress of each TouchableOpacity leads to a handlePress function.
- The handlePress function checks if a goToPath prop has been passed to the component. 
- If goToPath is defined, then the onPress will navigate to the path.
- If goToPath is NOT define, the onPress will change opacity but do nothing

### CODE REVIEW

File: src/screens/MyCar/DriversHomeScreen.js [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  TouchableOpacity,
+} from 'react-native';
+import DriversHomeScreenText from '../../assets/DriversHomeScreenText';
+import galleryData from '../../assets/serviceGalleryData';
+import ServiceCard from './Components/serviceCard';
+import BigServiceCard from './Components/BigServiceCard';
+import MessageCard from './Components/MessageCard';
+import messagesGalleryData from '../../assets/messagesGalleryData';
+
+const DriversHomeScreen = ({navigation}) => {
```
Comment (henaharon) [inline src/screens/MyCar/DriversHomeScreen.js:20]: I would have separate all the code to smaller components

File: src/screens/MyCar/DriversHomeScreen.js [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  ScrollView,
+  Image,
+  Dimensions,
+  Text,
+  FlatList,
+  TouchableOpacity,
+} from 'react-native';
+import DriversHomeScreenText from '../../assets/DriversHomeScreenText';
+import galleryData from '../../assets/serviceGalleryData';
+import ServiceCard from './Components/serviceCard';
```
Comment (henaharon) [inline src/screens/MyCar/DriversHomeScreen.js:15]: i get an error, Error: Unable to resolve module ./Components/serviceCard from /Users/henaharon/Desktop/MyCarApp/src/screens/MyCar/DriversHomeScreen.js:

I think it׳s the wrong import

File: src/screens/MyCar/DriversHomeScreen.js [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+import BigServiceCard from './Components/BigServiceCard';
```
Comment (henaharon) [inline src/screens/MyCar/DriversHomeScreen.js:16]: I think it׳s the wrong import

File: src/screens/MyCar/DriversHomeScreen.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+import MessageCard from './Components/MessageCard';
```
Comment (henaharon) [inline src/screens/MyCar/DriversHomeScreen.js:17]: I think it׳s the wrong import

File: src/screens/MyCar/DriversHomeScreen.js [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+          style={styles.backgroundImage}
+          resizeMode="cover"
+        />
+      </View>
+      <ScrollView contentContainerStyle={styles.scrollContainer}>
+        <View style={styles.content}>
+          {/* Transparent Logo */}
+          <View style={styles.logoContainer}>
+            <Image
+              source={require('../../assets/B1-assets/elementsLogosMyCarFullVerticalColorBlack.png')}
```
Comment (henaharon) [inline src/screens/MyCar/DriversHomeScreen.js:49]: icons should be in uiKit/icons
pls move all icons to uiKit/icons

File: src/screens/MyCar/b1Components/ServiceCard.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  const handlePress = () => {
+    if (goToPath) {
+      navigation.navigate(goToPath);
+    } else {
+      // Stay in the same page
+    }
+  };
+
+  return (
+    <TouchableOpacity onPress={handlePress} activeOpacity={0.8}>
```
Comment (henaharon) [inline src/screens/MyCar/b1Components/ServiceCard.js:14]: there is no shadow as showed in design

File: src/screens/MyCar/DriversHomeScreen.js (hunk 2 above)
Comment (sufka861) [reply src/screens/MyCar/DriversHomeScreen.js:15]: ok fixed

File: src/screens/MyCar/DriversHomeScreen.js (hunk 3 above)
Comment (sufka861) [reply src/screens/MyCar/DriversHomeScreen.js:16]: fixed

File: src/screens/MyCar/DriversHomeScreen.js (hunk 4 above)
Comment (sufka861) [reply src/screens/MyCar/DriversHomeScreen.js:17]: fixed

File: src/screens/MyCar/DriversHomeScreen.js (hunk 5 above)
Comment (sufka861) [reply src/screens/MyCar/DriversHomeScreen.js:49]: fixed

File: src/screens/MyCar/b1Components/ServiceCard.js (hunk 6 above)
Comment (sufka861) [reply src/screens/MyCar/b1Components/ServiceCard.js:14]: fixed

**Review by henaharon — APPROVED** [review]

**Pull request merged**